internal/types/types_scan_result.go

```diff
@@ -57,6 +57,15 @@ func (r *ScanResult) SetOpenssl(info OpensslInfo) *ScanResult {
 	return r
 }
 
+func (r *ScanResult) SetOS(info OSInfo) *ScanResult {
+	if info.Error != nil {
+		r.SetError(ErrDistributionFileMissing)
```

Comment:
This kind of hides the underlying error. If it is at least logged, it would be OK, but it is not.

I think some images doesn't have `/etc/redhat-release` and maybe they need to be excluded for this error. WDYT @mrunalp ?

Reply:
Yeah - I think that's probably case with scratch images. Curious to know if they should be excluded or if we just propose they're not used at all?

Reply:
Yes, we need to support exclusions for this type of error like we have for other error types. 
Also, we should include that exclusion for the ubi9 and other known failures as part of merging this PR so we don't break existing CI flows.

Reply:
The issue that we still lose the underlying error from `os.ReadFile()` is still not solved.

Reply:
For this are you looking for a log message, or something added directly to the error we pass through in the report?

Reply:
Since `open` or `read` may fail for many reasons, I guess that we should only translate ENOENT (aka `os.ErrNotExist`) into `os.ErrDistributionFileMissing`, for the sake of being able to exclude it, and report all other errors (as they are not expected) as-is.

Reply:
By "report" I mean "add it to scan results".